src/content/docs/dev/send-tokens/interchain-tokens/integrate-custom-token.mdx

```diff
@@ -39,42 +39,79 @@ contract MyInterchainToken is ERC20, ERC20Burnable {
+    address tokenAddress, // your custom token address
+    TokenManagerType tokenManagerType, // token manager type
+    address operator // minter address
+) external payable returns (bytes32 tokenId) {};
 ```
 
-This function deploys a Token Manager for your token, connects it to ITS upon deployment, and returns the unique interchain token ID.
+This function registers an existing ERC-20 token (your custom token) under a `tokenId` computed from the provided `salt`. A token metadata registration message will also be sent to the ITS Hub. Then, the token can be linked to remote tokens on different chains by submitting the `linkToken` function from the same `msg.sender` using the same `salt`.
 
-## Assign the `minter` Role to Your Token's Token Manager
+### Link Custom Token with the Interchain Token Factory
 
-Transfer the minter role to the Token Manager using the `transferMintership` function:
+To link a custom token, use the [`linkToken`](https://github.com/axelarnetwork/interchain-token-service/blob/94677f1072da127a0ee7ff5ab6784dcfe11789e8/contracts/InterchainTokenFactory.sol#L541) function to link your custom tokens on your respective chains.
 
 ```solidity
-myToken.transferMintership(tokenManagerAddress);
+function linkToken(
+    bytes32 salt, // your unique salt value previously used
+    string calldata destinationChain, // destination chain
+    bytes calldata destinationTokenAddress, // destination custom token address
+    TokenManagerType tokenManagerType, // token manager type
+    bytes calldata linkParams, // minter address
+    uint256 gasValue // gas value
+) external payable returns (bytes32 tokenId) {}
 ```
 
-The [`transferMintership()`](https://github.com/axelarnetwork/interchain-token-service/blob/main/contracts/utils/Minter.sol#L31) function transfers the minter role to the Token Manager. This is necessary because when your token is bridged into a destination chain, the `msg.sender` of the `mint()` call will be the Token Manager.
+This function links a remote token on `destinationChain` to a local token corresponding to the `tokenId` computed from the provided `salt`.
 
-## Send Token using Interchain Transfer Function
+> **Note:** A local token must be registered first using the `registerCustomToken` function.
 
-An Interchain Token is an ERC-20 that is connected to [ITS](https://github.com/axelarnetwork/interchain-token-service/blob/main/contracts/InterchainTokenService.sol) upon deployment comes built-in with the `interchainTransfer()` function, which allows users to bridge their token between any blockchain to which it is deployed. Take a look at the diagram below to see how the `interchainTransfer()` function works.
+### Assign the `minter` Role to Your Token's Token Manager
+
+Transfer the minter role to the Token Manager using the `transferMintership` function on your token. To retrieve your token manager address, use the `tokenManagerAddress` function on the ITS contract by specifying your `tokenId`.
```

Comment:
Perhaps worth linking to the Minter contract where there this is defined
https://github.com/axelarnetwork/interchain-token-service/blob/v/contracts/utils/Minter.sol#L31

Also could be good to add a note that this it is not required to pass minter role by triggering this function as certain projects may have a separate Access Controls contract where they would interact with instead.

Reply:
added

Reply:
I added the reference alone since minter transfer can be done in multiple ways depending on their token. So this is just an example.